lib/auth.coffee

```diff
@@ -215,6 +216,43 @@ getAuth = (deps, opts) ->
 		.catchReturn(false)
 		.asCallback(callback)
 
+	###*
+	# @summary Creates a new API key for user
+	# @name createApiKey
+	# @public
+	# @function
+	# @memberof resin.auth
+	#
+	# @description This method registers a new api key for the current user with the name given.
+	#
+	# @param {String} name - the API key name
+	#
+	# @fulfil {String} - API key
+	# @returns {Promise}
+	#
+	# @example
+	# resin.auth.createApiKey(apiKeyName).then(function(apiKey) {
+	# 	console.log(apiKey);
+	# });
+	#
+	# @example
+	# resin.auth.createApiKey(apiKeyName, function(error, apiKey) {
+	# 	if (error) throw error;
+	# 	console.log(apiKey);
+	# });
+	###
+	exports.createApiKey = (name, callback) ->
+		request.send
+			method: 'POST'
+			url: '/api-key/user/full'
+			baseUrl: apiUrl
+			body:
+				name: name
+		.get('body')
+		.catch ->
```

Comment:
How about if the server returns a 500?
Oh, I see that we are doing the same in the rest of the auth methods! We might want to consider narrowing all these `.catch()`es in a separate PR.

Reply:
Issue filed: https://github.com/resin-io/resin-sdk/issues/462